project-base/storefront/components/Pages/HomePage/ExchangeTokenHandler.tsx

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+import { useLoginViaExchangeToken } from 'utils/auth/useLoginViaExchangeToken';
+import { showErrorMessage } from 'utils/toasts/showErrorMessage';
+
+export const ExchangeTokenHandler: FC = () => {
+    const router = useRouter();
+    const loginViaExchangeToken = useLoginViaExchangeToken();
+    const processedTokens = useRef(new Set<string>());
+
+    useEffect(() => {
```

Comment:
I would like to hear an opinion of one of FE devs. 
I have expected use of middleware to handle the login, but maybe it's not possible 🤷‍♂️

And with the upcoming approuter we should probably avoid use of useEffect.

Reply:
maybe the review will become re-implementation actually :smile: